frontends/p4/simplifyParsers.cpp

```diff
@@ -55,6 +55,11 @@ class RemoveUnreachableStates : public Transform {
         if (state->name == IR::ParserState::start ||
             state->name == IR::ParserState::reject)
             return state;
+        // Do not eliminate states with annotations other than name.
+        if (state->getAnnotations()->annotations.size() > 1
```

Comment:
The comment does not reflect what the code is doing. There is no check that there is any annotation that is not "name". Note that nothing prevents an object having multiple annotations named in the same way.